fiona/_transform.pyx

```diff
@@ -224,6 +199,47 @@ def _transform_geom(
                         inner_coords.append(list(zip(xp, yp)))
                 new_coords.append(inner_coords)
 
-        g['coordinates'] = new_coords
+        out_geom['coordinates'] = new_coords
+    return out_geom
+
+
+def _transform_geom(
+        src_crs, dst_crs, geom, antimeridian_cutting, antimeridian_offset,
+        precision):
+    """Return a transformed geometry."""
+    cdef char *proj_c = NULL
+    cdef char *key_c = NULL
+    cdef char *val_c = NULL
+    cdef char **options = NULL
+    cdef OGRSpatialReferenceH src = NULL
+    cdef OGRSpatialReferenceH dst = NULL
+    cdef void *transform = NULL
+    cdef OGRGeometryFactory *factory = NULL
+    if not all([src_crs, dst_crs]):
+        raise RuntimeError("Must provide a source and destination CRS.")
+    src = _crs_from_crs(src_crs)
+    dst = _crs_from_crs(dst_crs)
+    transform = _crs.OCTNewCoordinateTransformation(src, dst)
+    # Transform options.
+    options = _csl.CSLSetNameValue(
+                options, "DATELINEOFFSET", 
+                str(antimeridian_offset).encode('utf-8'))
+    if antimeridian_cutting:
+        options = _csl.CSLSetNameValue(options, "WRAPDATELINE", "YES")
+
+    factory = new OGRGeometryFactory()
 
-    return g
+    if isinstance(geom, DICT_TYPES):
```

Comment:
@snowman2 this won't work in 1.9 because we're introducing a Geometry class that is not a dict. Instead, let's see if geom is iterable using either `isinstance(geom, Iterable)` like in https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterable or `try: geoms = iter(geom); except TypeError: geoms = [geom]`.

Reply:
Hmmm, I looked at your PR adding the Geometry class and I think that the Geometry class would be considered an iterable. Maybe this can be updated to check for the Geometry instance specifically?

Reply:
Right. How about `if isinstance(geom, collections.abc.Mapping)` then? The OGRGeomBuilder class gets the type and coordinates of `geom` using only its mapping interface.

Reply:
Actually, that should be accounted for with `DICT_TYPES`

https://github.com/Toblerity/Fiona/blob/d6ec15f1c2e7a860bc57d35a51e36aeb48377a1a/fiona/compat.py#L26

In that case, is it okay to leave it as is?

Reply:
Can't believe I overlooked that. Yes, this is fine.

Reply:
Sounds good 👍. I will work on adding tests and updating the docstrings later.